### DISCUSSION TITLE
Remotely trigger test

### DISCUSSION THREAD

**Discussion by twistedsanity** [alexjustesen/speedtest-tracker#1046] [Q&A] [answered]
This has been a great tool for monitoring my internet connection and ensuring I am getting the service I pay for. However due to streaming, kids playing games, etc... my results are sometime unfairly low (due to consumption not service). I had an idea to make it work more like a 'samknows whitebox' in that it only runs tests when throughput is low/idle. To do this I am currently using Home Assistant with pfsense integration, I set up a trigger of every 10 mins, but before executing it checks to see that traffic up/down is below a defined level, and if not delay. But so far have been unsuccessful in finding how to trigger the test.

I hoped there would be an easy method such as api/http call or similar that will trigger a manual speedtest immediately.

I have been trying to mimic the button press manually but as yet unable to figure it out.

**Comment by alexjustesen** [comment, marked as answer]
Currently no, but I think this is a fantastic idea and something that could extend how tests are scheduled as well.

**Reply by alexjustesen** [reply]
Additionally you can follow https://github.com/alexjustesen/speedtest-tracker/issues/1047 to see when this is implemented.

**Comment by metaathron** [comment]
I know, this thread is old, but you can now trigger tests from Home Assistant also using my integration: https://github.com/metaathron/ha-speedtest-tracker
